Gordon360/Services/HousingService.cs

```diff
@@ -635,58 +611,54 @@ public ApartmentApplicationViewModel GetApartmentApplication(int applicationID)
                 List<ApartmentApplicantViewModel> applicantsList = new List<ApartmentApplicantViewModel>();
                 foreach (GET_AA_APPLICANTS_BY_APPID_Result applicantDBModel in applicantsResult)
                 {
-                    StudentProfileViewModel student = Data.StudentData.FirstOrDefault(x => x.AD_Username.ToLower() == applicantDBModel.Username.ToLower());
-                    if (student != null)
-                    {
-                        // If the student information is found, create a new ApplicationViewModel and fill in its properties
-                        ApartmentApplicantViewModel applicantModel = new ApartmentApplicantViewModel
-                        {
-                            ApplicationID = applicationID,
-                            Profile = student,
-                            Username = student.AD_Username,
-                            Age = null, // Not yet implemented
-                            Class = student.Class,
-                            OffCampusProgram = applicantDBModel.AprtProgram,
-                            Probation = false, // Not yet implemented. This is where we will put the code to check if a student has a probation
-                            Points = 0, // Initialize to zero. The point actual points are calculated a few lines below this
-                        };
-                        // The probation data is already in the database, we just need to write a stored procedure to get it
+                    ApartmentApplicantViewModel applicantModel = applicantDBModel; //implicit conversion
 
-                        // Calculate application points
-                        int points = 0;
-                        switch (applicantModel.Class)
+                    // If the student information is found, create a new ApplicationViewModel and fill in its properties
+                    if (applicantModel.Profile != null && applicantDBModel.AprtAppID == applicationID)
+                    {
+                        // Only add the birthdate, probabtion, and points if the user is authorized to view that information
+                        if (true) // if the current user is a housing admin or super admin 
```

Comment:
Is 'true' a placeholder?

Reply:
yes, it is very much a placeholder 😆

Reply:
I wanted to ask @EjPlatzer  about what would be the best way to make that check possible. I can think of a few ideas, but they all seem needlessly convoluted and inefficient

Reply:
I just thought of something new: I think we can make a helper method and give it a `StateYourBusiness`, I saw that used in the AcountService. However, if `StateYourBusiness` rejects with an error, then we would need to somehow catch that, since we don't actually want it to throw an error, we just want to skip the birthdate part if the user is not authorized to that info

Reply:
Actually I believe `StateYourBusiness` only runs on Controllers. Any `StateYourBusiness` attributes in Services were put there mistakenly.

I think the simplest thing to do would be to pass down the `ID` of the user requesting the Application from the Controller, and then you can `CheckIfHousingAdmin(requestorID)` and `_adminService.Get(requestorID)`. You'd have to catch the exceptiong thrown by the Admin service, but that shouldn't be too hard to do.

Reply:
Yes, on second thought putting the authorization logic in the controller is much more reasonable.

That also means that if the service method is ever called from elsewhere in the API, you're only passing it the `canViewBirthdate` flag instead of having to invent an ID for a hypothetical user if the API needed internal access to that info. An unlikely use case, but still useful in case it was ever needed.

Reply:
I just pushed an update of this change